src/Psl/Option/Option.php

```diff
@@ -305,4 +306,78 @@ public function equals(mixed $other): bool
     {
         return Comparison\equal($this, $other);
     }
+
+    /**
+     * Combines two `Option` values into a single `Option` containing a tuple of the two inner values.
+     * If either of the `Option`s is `None`, the resulting `Option` will also be `None`.
+     *
+     * @note: If an element is `None`, the corresponding element in the resulting tuple will be `None`.
```

Comment:
I'm a bit confused here. This is not true right?
If either of them is None, the result is None.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b59993df008ad0c70f0a4c979f1e5dfb

Reply:
![goddamn-right-yes](https://github.com/azjezz/psl/assets/1777519/61352926-ede4-4c59-b4c4-1ee16c557cbe)